### PR TITLE
Add firewall rules and ing class clarifications

### PR DESCRIPTION
**Pull request by pdoreau**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1672)

### CODE REVIEW

File: ingress/controllers/nginx/README.md [hunk 1]
```diff
@@ -78,6 +78,8 @@ Next we create a couple of Ingress rules
 kubectl create -f examples/ingress.yaml
 ```
 
+If your version of Kubernetes is higher than 1.3, you can restrict the ingress usage to NGINX with this annotation :`kubernetes.io/ingress.class: "nginx"`
```
Comment (bprashanth) [inline ingress/controllers/nginx/README.md]: please surface this in the requirements section instead, something like: 
```
if you're running multiple ingress controllers, or running on a cloudprovider that natively handles 
ingress, you need to specify the annotation `kubernetes.io/ingress.class: "nginx"` in all ingresses 
that you would like this controller to claim. Not specifying the annotation will lead to multiple 
ingress controllers claiming the same ingress. Specifying the wrong value will result in all ingress 
controllers ignoring the ingress. Multiple ingress controllers running in the same cluster was not 
supported in Kubernetes versions < 1.3.
```

File: ingress/controllers/nginx/README.md [hunk 2]
```diff
@@ -102,6 +104,7 @@ Check NGINX it is running with the defined Ingress rules:
 $ LBIP=$(kubectl get node `kubectl get po -l name=nginx-ingress-lb --template '{{range .items}}{{.spec.nodeName}}{{end}}'` --template '{{range $i, $n := .status.addresses}}{{if eq $n.type "ExternalIP"}}{{$n.address}}{{end}}{{end}}')
 $ curl $LBIP/foo -H 'Host: foo.bar.com'
 ```
+You may need to add a firewall rule to allow traffic on port 80. This is related to your cloud provider.
```
Comment (bprashanth) [inline ingress/controllers/nginx/README.md]: and 443 right

Comment (bprashanth) [inline ingress/controllers/nginx/README.md]: Oh, I see you have it below. 

Hmm, how about just creating a different section:

## Running multiple ingress controllers 

If you're running multiple ingress controllers, or running on a cloudprovider that natively handles 
ingress, you need to specify the annotation `kubernetes.io/ingress.class: "nginx"` in all ingresses 
that you would like this controller to claim. Not specifying the annotation will lead to multiple 
ingress controllers claiming the same ingress. Specifying the wrong value will result in all ingress 
controllers ignoring the ingress. Multiple ingress controllers running in the same cluster was not 
supported in Kubernetes versions < 1.3.

## Running on Cloudproviders

If you're running this ingress controller on a cloudprovider, you should assume the provider also has a native 
Ingress controller and specify the `ingress.class` annotation as indicated in [this section](#running-multiple-ingress-controllers).
In addition to this, you will need to add a firewall rule for each port this controller is listening on, i.e :80 and :443.

**Comment by pdoreau** [comment]
@bprashanth fine to me, updated with your suggestion

File: ingress/controllers/nginx/README.md [hunk 3]
```diff
@@ -340,6 +342,21 @@ Please check the example `example/rc-default.yaml`
 
 To extract the information in JSON format the module provides a custom URL: `/nginx_status/format/json`
 
+### Running multiple ingress controllers
+
+If you're running multiple ingress controllers, or running on a cloudprovider that natively handles 
+ingress, you need to specify the annotation kubernetes.io/ingress.class: "nginx" in all ingresses
```
Comment (bprashanth) [inline ingress/controllers/nginx/README.md]: please quote this, ie `kubernetes.io/ingress.class: "nginx"`

**Comment by bprashanth** [comment]
LGTM just one nit

**Comment by bprashanth** [comment]
LGTM, please sign CLA

**Comment by pdoreau** [comment]
I signed it!

**Pull request merged**